pkg/prometheus/server/operator.go

```diff
@@ -1221,3 +1230,48 @@ func makeSelectorLabels(name string) map[string]string {
 		"prometheus":                    name,
 	}
 }
+
+func validateAlertmanagerEndpoints(am monitoringv1.AlertmanagerEndpoints) error {
+	var nonNilFields []string
+
+	//nolint:staticcheck // Ignore SA1019 this field is marked as deprecated.
+	if am.BearerTokenFile != "" {
+		nonNilFields = append(nonNilFields, fmt.Sprintf("%q", "bearerTokenFile"))
+	}
+
+	for k, v := range map[string]interface{}{
+		"basicAuth":     am.BasicAuth,
+		"authorization": am.Authorization,
+		"sigv4":         am.Sigv4,
+	} {
+		if reflect.ValueOf(v).IsNil() {
+			continue
+		}
+		nonNilFields = append(nonNilFields, fmt.Sprintf("%q", k))
+	}
+
+	if len(nonNilFields) > 1 {
+		return fmt.Errorf("%s can't be set at the same time, at most one of them must be defined", strings.Join(nonNilFields, " and "))
+	}
+
+	return nil
+}
+
+func addAlertmanagerEndpointsToStore(ctx context.Context, store *assets.StoreBuilder, namespace string, ams []monitoringv1.AlertmanagerEndpoints) error {
+	for i, am := range ams {
+		if err := validateAlertmanagerEndpoints(am); err != nil {
+			return fmt.Errorf("alertmanager %d: %w", i, err)
+		}
```

Comment:
In `pkg/prometheus/server/operator.go` we already call `validateAlertmanagerEndpoints` right before calling `addAlertmanagerEndpointsToStore`. Do we need to call it again here?

Reply:
I must've messed up the rebase, let me clear it up.